sys/include/net/sock/dtls.h

```diff
@@ -30,15 +30,26 @@
  * - Server operation
  *   1. Create UDP sock @ref sock_udp_create()
  *   2. Create DTLS sock @ref sock_dtls_create() using role
- *      @ref SOCK_DTLS_SERVER
- *   3. Start listening with @ref sock_dtls_recv()
+ *      @ref SOCK_DTLS_SERVER.
+ *   3. Optionally:
+ *          - when using PSK ciphersuites, set a hint @ref sock_dtls_set_psk_identity_hint()
+ *          - add extra credentials @ref sock_dtls_add_credential()
+ *          - when using ECC ciphersuites, set a callback for credential selection
+ *            @ref sock_dtls_set_rpk_cb()
+ *   4. Start listening with @ref sock_dtls_recv()
  * - Client operation
  *   1. Create UDP sock @ref sock_udp_create()
  *   2. Create DTLS sock @ref sock_dtls_create() using role
  *      @ref SOCK_DTLS_CLIENT
- *   3. Start handshake session to server @ref sock_dtls_session_init()
- *   4. Handle incoming handshake packets with @ref sock_dtls_recv()
- *   5. Send packet to server @ref sock_dtls_send()
+ *   3. Optionally:
+ *          - add extra credentials @ref sock_dtls_add_credential()
+ *          - when using PSK ciphersuites, set a callback for hint reception and credential
```

Comment:
Change order of first and second bullet to match the above server operations, or does this break the client logic?

Reply:
Isn't it matching the server order? 

0. set hint (not available in client)
1. add extra credentials
2. set PSK callback (not available in server)
3. set RPK callback